merge/conflict_test.go

```diff
@@ -92,3 +92,23 @@ func TestToSet(t *testing.T) {
 		t.Fatalf("expected\n%v\n, but got\n%v\n", expected, actual)
 	}
 }
+
+func TestConflictsFromManagers(t *testing.T) {
+
+	got := merge.ConflictsFromManagers(fieldpath.ManagedFields{
+		"Bob": fieldpath.NewVersionedSet(
+			_NS(
+				_P("obj", "template", "obj", "list", _KBF("name", "a"), "id"),
+				_P("obj", "template", "obj", "list", _KBF("name", "a"), "key"),
```

Comment:
Are these the simplest form that trigger the bug?!

Reply:
yes, this is the simplest which triggers the bug.

Reply:
@apelisse can you please review again when you get a chance